bin/strata-cli/src/link.rs

```diff
@@ -0,0 +1,184 @@
+use std::fmt::Display;
+
+use alloy::primitives::{Address as StrataAddress, TxHash};
+use bdk_wallet::bitcoin;
+
+/// A thing that exists or is represented onchain.
+/// Currently, this is either a transaction or an address.
+pub enum OnchainObject<'a> {
+    Transaction(Txid<'a>),
+    Address(Address<'a>),
+}
+
+impl<'a> From<&'a bitcoin::Address> for OnchainObject<'a> {
+    fn from(address: &'a bitcoin::Address) -> Self {
+        OnchainObject::Address(Address::Bitcoin(address))
+    }
+}
+
+impl<'a> From<&'a StrataAddress> for OnchainObject<'a> {
+    fn from(address: &'a StrataAddress) -> Self {
+        OnchainObject::Address(Address::Strata(address))
+    }
+}
+
+impl<'a> From<&'a bitcoin::Txid> for OnchainObject<'a> {
+    fn from(txid: &'a bitcoin::Txid) -> Self {
+        OnchainObject::Transaction(Txid::Bitcoin(txid))
+    }
+}
+
+impl<'a> From<&'a TxHash> for OnchainObject<'a> {
+    fn from(txid: &'a TxHash) -> Self {
+        OnchainObject::Transaction(Txid::Strata(txid))
+    }
+}
+
+impl<'a> OnchainObject<'a> {
+    /// Create a link to the object on the given explorer.
+    /// Should be of the form `http{s}://{domain}`.
+    pub fn with_explorer<'b>(self, explorer: &'b str) -> Link<'a, 'b> {
+        Link {
+            object: self,
+            explorer_ep: explorer,
+        }
+    }
+
+    /// Create a link to the object on the given explorer, if it exists.
+    /// Should be of the form `http{s}://{domain}`.
+    ///
+    /// If `explorer` is `None`, the object will be represented as-is.
+    ///
+    /// This is primarily a helper for displaying an OnChainObject in a user-facing context.
+    pub fn with_maybe_explorer<'b>(self, explorer: Option<&'b str>) -> MaybeLink<'a, 'b> {
+        match explorer {
+            Some(dmn) => MaybeLink::Link(self.with_explorer(dmn)),
+            None => MaybeLink::Object(self),
+        }
+    }
+}
+
+/// A helper trait for pretty printing something into a human-readable string.
+/// Differs from `Display` in that it might add other wording outside of purely
+/// the string representation of `self`.
+///
+/// E.g. a transaction might be prefixed with "Transaction: " or "Tx: ".
+pub trait PrettyPrint {
+    fn pretty(&self) -> String;
+}
+
+/// A helper enum for pretty printing something that might be a link or an object.
+pub enum MaybeLink<'a, 'b> {
+    Link(Link<'a, 'b>),
+    Object(OnchainObject<'a>),
```

Comment:
docstrings in the variants.

Reply:
Yes, it is but I think we should lean towards more verbose documentation and add docstrings to anything that is `pub` (and some intricate not `pub`) stuff.